### PR TITLE
Send startup event on LS start

### LINKED ISSUE #725: Combine yaml.server.initialized event with the startup event
`startup` and `yaml.server.initialized` events represent 90+% of the telemetry events we send. Since each event has a literal cost and yaml.server.initialized doesn't bring much information, we should simply include the yaml.server.initialized  info into the startup event and send just that one.

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
Removes `yaml.server.initialized` and send `startup` instead 

### What issues does this PR fix or reference?
Resolve #725

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -176,8 +177,7 @@ export function startClient(
     client.onRequest(FSReadFile.type, (fsPath: string) => {
       return workspace.fs.readFile(Uri.file(fsPath)).then((uint8array) => new TextDecoder().decode(uint8array));
     });
-
-    runtime.telemetry.send({ name: 'yaml.server.initialized' });
+    runtime.telemetry.sendStartupEvent();
```
Comment (fbricon) [inline src/extension.ts:180]: can we send a boolean, for (un)successful startup?, and an error message in case of unsuccessful startup

Comment (evidolob) [reply src/extension.ts:180]: the `sendStartupEvent` doesn't have any parameters, we already send all LS errors, including startup errors.
Or you mean, do not use `sendStartupEvent` and instead send some   `{ name: 'yaml.server.initialized', successful : true/false }` ?

Comment (evidolob) [reply src/extension.ts:180]: the `sendStartupEvent` doesn't have any parameters, we already send all LS errors, including startup errors.
Or you mean, do not use `sendStartupEvent` and instead send some   `{ name: 'yaml.server.initialized', successful : true/false }` ?

Comment (evidolob) [reply src/extension.ts:180]: the `sendStartupEvent` doesn't have any parameters, we already send all LS errors, including startup errors.
Or you mean, do not use `sendStartupEvent` and instead send some   `{ name: 'yaml.server.initialized', successful : true/false }` ?

**Review by gorkem — APPROVED** [review]

**Review by msivasubramaniaan — APPROVED** [review]

File: src/extension.ts (hunk 1 above)
Comment (evidolob) [reply src/extension.ts:180]: @fbricon WDYT?

Comment (fbricon) [reply src/extension.ts:180]: You can manually craft a startup event like:
```json
{ 
name: 'startup', 
 properties: {
   'yaml.server.initialized' : true/false,
   'error': the error message if the server failed to initialize
 }
}
```

File: src/extension.ts [hunk 2]
```diff
@@ -257,6 +263,19 @@ function getSchemaAssociations(): ISchemaAssociation[] {
...
 
+function sendStartupTelemetryEvent(telemetry: TelemetryService, initialized: boolean, err?: Error): void {
+  const startUpEvent = {
+    name: 'startup',
+    properties: {
+      'yaml.server.initialized': initialized,
+    },
+  };
+  if (err) {
+    startUpEvent.properties['error'] = err;
```
Comment (fbricon) [inline src/extension.ts:274]: it doesn't work unless err is a string. You need to check if err.message exists and send it

Comment (fbricon) [inline src/extension.ts:274]: it doesn't work unless err is a string. You need to check if err.message exists and send it

File: src/extension.ts [hunk 3]
```diff
@@ -257,6 +263,19 @@ function getSchemaAssociations(): ISchemaAssociation[] {
...
+  }
```
Comment (fbricon) [inline src/extension.ts:275]:
```typescript
  if (err?.message) {
    startUpEvent.properties['error'] = err.message;
  }
```

**Review by fbricon — APPROVED** [review]

**Pull request merged**